SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+				if ($NSXToken) {
+					$nsxtEdgeNodes = (Get-NsxtEdgeCluster | Where-Object {$_.member_node_type -eq "EDGE_NODE"})
+					foreach ($nsxtEdgeNode in $nsxtEdgeNodes.members) {
+						$nsxtConfigTypeValues =  New-Object -TypeName psobject
+						$nsxtConfigTypeValues | Add-Member -notepropertyname "transportNodeID" -notepropertyvalue "Empty"
+						$nsxtConfigTypeValues | Add-Member -notepropertyname "passwdMinimumLength" -notepropertyvalue "Empty"
+						$nsxtConfigTypeValues | Add-Member -notepropertyname "cliPasswdMaxFailAttempts" -notepropertyvalue "Empty"
+						$nsxtConfigTypeValues | Add-Member -notepropertyname "cliPasswdMaxFailIntervalInSec" -notepropertyvalue "Empty"
+
+						$response = Get-NsxtEdgeNodeAuthPolicy -nsxtManager $server -nsxtEdgeNodeID $nsxtEdgeNode.transport_node_id
+						$nsxtConfigTypeValues.transportNodeID = $nsxtEdgeNode.transport_node_id
+						$nsxtConfigTypeValues.passwdMinimumLength = $response.minimum_password_length
+						$nsxtConfigTypeValues.cliPasswdMaxFailAttempts = $response.cli_max_auth_failures
+						$nsxtConfigTypeValues.cliPasswdMaxFailIntervalInSec = $response.cli_failed_auth_lockout_period  
+
+						$nsxtEdgeNodeAdds = GetNsxtEdgeNode -transportNodeID $nsxtEdgeNode.transport_node_id
+						$nodeConfigValue = New-Object -TypeName psobject
+						$nodeConfigValue | Add-Member -notepropertyname "FQDN" -notepropertyvalue $nsxtEdgeNodeAdds.node_deployment_info.node_settings.hostname
+						$nodeConfigValue | Add-Member -notepropertyname "productType" -notepropertyvalue "NSXEdge"
+						$nodeConfigValue | Add-Member -notepropertyname "productAttributes" -notepropertyvalue $nsxtConfigTypeValues
+						$passwordPolicyNodes.Add($nodeConfigValue)
+						Remove-Variable -Name nodeConfigValue
+						Remove-Variable -Name nsxtConfigTypeValues
+					}
+					return $passwordPolicyNodes
+				} else {
+					if ($ErrorLog -match '"error_code":404') {
+						$count += 1
+						Write-LogMessage -Type WARNING -Message "Unable to submit request. Retry again in 20 secs (Try $count/3)"
+						Start-Sleep -s 20
+					} else {
+						Write-LogMessage -Type INFO -Error "Failed to obtain access token from NSX-T server '$server',  please verify your credentials and try again"  -ErrorAction Stop
+					}
+				}
+			}
+			Write-LogMessage -Type Error -Message "Failed to obtain access token from NSX-T server '$server',  please verify your credentials and try again" -ErrorAction Stop
+		} else {
+				Write-LogMessage -Type Error -Message "Testing a connection to server $server failed, please check your details and try again" -ErrorAction Stop
+		}
+    }
+    Catch {
+        Debug-ExceptionWriter -object $_
+    }
+	Finally {
+		Write-LogMessage -Type INFO -Message "Retrieving NSX-T Edge Node password policies completed"
+	}
+}
+
+Function GetWSAPasswordPolicyAll {
+	<#
+        .SYNOPSIS
+        Retrieve Workspace One Access servers' Password Policy
```

Comment:
"ONE"

Reply:
addressed in 99029d5ec8015e0c5c645e9abc7c3384e1b3091e